### PR TITLE
[zh-cn] update intersection observer api

### PR DESCRIPTION
**Pull request by jasonren0403**

### CODE REVIEW

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 1]
```diff
@@ -1,43 +1,45 @@
 ---
-title: Intersection Observer API
+title: 交叉观察器 API
```
Comment (jasonren0403) [inline files/zh-cn/web/api/intersection_observer_api/index.md:2]: 交叉观察器：https://www.chinaw3c.org/archives/1964/

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 2]
```diff
@@ -90,40 +93,48 @@ let callback = (entries, observer) => {
...
-- 第二个盒子只有唯一的值 `[0.5]`。
-- 第三个盒子的 thresholds 按 10% 从 0 递增 (0%, 10%, 20%, etc.)。
-- 最后一个盒子为 `[0, 0.25, 0.5, 0.75, 1.0]`。
+请注意，可能不存在相交矩形，如果交集正好沿着两者之间的边界，或者 {{domxref("IntersectionObserverEntry.boundingClientRect", "boundingClientRect")}} 的面积为零。这种目标和根共享边界线的状态不足以被视为过渡到相交状态。
+
+为了让我们感受下阈值是如何工作的，尝试滚动以下的盒子，每一个带颜色盒子的四个边角都会展示自身在根元素中的可见程度百分比，所以在你滚动根元素的时候你将会看到四个边角的数值一直在发生变化。每一个盒子都有不同的阈值：
+
+- 第一个盒子为每个可见度百分点设置了一个阈值；也就是说，{{domxref("IntersectionObserver.thresholds")}} 数组为 `[0.00, 0.01, 0.02, /*...,*/ 0.99, 1.00]`。
+- 第二个盒子只有一个阈值，位于 50% 刻度处。
+- 第三个盒子每隔 10%的可见度设置一个阈值（0%、10%、20% 等）。
```
Comment (jasonren0403) [inline files/zh-cn/web/api/intersection_observer_api/index.md:136]:
```suggestion
- 第三个盒子每隔 10% 的可见度设置一个阈值（0%、10%、20% 等）。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 3]
```diff
@@ -506,21 +578,21 @@ function handleIntersect(entries, observer) {
 }
 ```
 
-For each {{domxref("IntersectionObserverEntry")}} in the list `entries`, we look to see if the entry's {{domxref("IntersectionObserverEntry.intersectionRatio", "intersectionRatio")}} is going up; if it is, we set the target's {{cssxref("background-color")}} to the string in `increasingColor` (remember, it's `"rgba(40, 40, 190, ratio)"`), replaces the word "ratio" with the entry's `intersectionRatio`. The result: not only does the color get changed, but the transparency of the target element changes, too; as the intersection ratio goes down, the background color's alpha value goes down with it, resulting in an element that's more transparent.
+对于 `entries` 列表中的每个 {{domxref("IntersectionObserverEntry")}}，我们都会查看该条目中的 {{{domxref("IntersectionObserverEntry.intersectionRadio", "intersectionRatio"}} 是否上升，如果是，我们就将目标的 {{cssxref("background-color")}} 设置为 `increasingColor` 中的字符串（记住，是 `"rgba(40, 40, 190, ratio)"`），并用条目的 `intersectionRatio` 替换“ratio”。结果是：不仅颜色发生了变化，目标元素的透明度也发生了变化；随着交集比率的降低，背景色的 alpha 值也会随之降低，从而使元素更加透明。
```
Comment (jasonren0403) [inline files/zh-cn/web/api/intersection_observer_api/index.md:581]:
```suggestion
对于 `entries` 列表中的每个 {{domxref("IntersectionObserverEntry")}}，我们都会查看该条目中的 {{domxref("IntersectionObserverEntry.intersectionRadio", "intersectionRatio")}} 是否上升，如果是，我们就将目标的 {{cssxref("background-color")}} 设置为 `increasingColor` 中的字符串（记住，是 `"rgba(40, 40, 190, ratio)"`），并用条目的 `intersectionRatio` 替换“ratio”。结果是：不仅颜色发生了变化，目标元素的透明度也发生了变化；随着交集比率的降低，背景色的 alpha 值也会随之降低，从而使元素更加透明。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 4]
```diff
@@ -320,37 +328,37 @@ intersectionCallback(entries) => {
...
-## A simple example
+## 一个简单的示例
 
-This simple example causes a target element to change its color and transparency as it becomes more or less visible. At [Timing element visibility with the Intersection Observer API](/zh-CN/docs/Web/API/Intersection_Observer_API/Timing_element_visibility), you can find a more extensive example showing how to time how long a set of elements (such as ads) are visible to the user and to react to that information by recording statistics or by updating elements..
+这个简单的示例会使目标元素在可见度变高或变低时改变颜色和透明度。在[使用交叉观察器 API 为元素可见性计时](/zh-CN/docs/Web/API/Intersection_Observer_API/Timing_element_visibility)中，可以找到一个更广泛的示例，展示如何对一组元素（如广告）对用户的可见时间进行计时，并通过记录统计数据或更新元素对该信息做出反应。
 
 ### HTML
 
-The HTML for this example is very short, with a primary element which is the box that we'll be targeting (with the creative ID `"box"`) and some contents within the box.
+这个示例的 HTML 代码非常简短，主要元素是我们要针对的方框（其 ID 为 `"box"`），以及方框中的一些内容。
```
Comment (jasonren0403) [inline files/zh-cn/web/api/intersection_observer_api/index.md:347]:
```suggestion
这个示例的 HTML 代码非常简短，主要元素是我们要针对的盒子（其 ID 为 `"box"`），以及盒子中的一些内容。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 5]
```diff
@@ -49,36 +51,37 @@ let options = {
...
+创建一个观察器后，需要给定一个目标元素进行观察。
 
 ```js
 let target = document.querySelector("#listItem");
 observer.observe(target);
-```
 
-每当目标满足该 IntersectionObserver 指定的 threshold 值，回调被调用。
+// 我们为观察者设置的回调将在第一次执行，
+// 它将等待我们为观察者分配目标（即使目标当前不可见）
```
Comment (jasonren0403) [inline files/zh-cn/web/api/intersection_observer_api/index.md:76]:
```suggestion
// 我们为观察器设置的回调将在第一次执行，
// 它将等待我们为观察器分配目标（即使目标当前不可见）
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 6]
```diff
@@ -49,36 +51,37 @@ let options = {
...
+```
 
-只要目标满足为 IntersectionObserver 指定的阈值，就会调用回调。回调接收 {{domxref("IntersectionObserverEntry")}} 对象和观察者的列表：
+每当目标满足该 `IntersectionObserver` 指定的 threshold 值，回调被调用。回调接收 {{domxref("IntersectionObserverEntry")}} 对象和观察者的列表：
```
Comment (jasonren0403) [inline files/zh-cn/web/api/intersection_observer_api/index.md:79]:
```suggestion
每当目标满足该 `IntersectionObserver` 指定的 threshold 值，回调被调用。回调接收 {{domxref("IntersectionObserverEntry")}} 对象和观察器的列表：
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 7]
```diff
@@ -1,43 +1,45 @@
...
-## Intersection observer 的概念和用法
+## 交叉观察器的概念和用法
 
-Intersection Observer API 允许你配置一个回调函数，当以下情况发生时会被调用
+交叉观察器 API 允许你配置一个回调函数，当以下情况发生时会被调用：
 
-- 每当目标 (**target**) 元素与设备视窗或者其他指定元素发生交集的时候执行。设备视窗或者其他元素我们称它为根元素或根 (**root**)。
-- Observer 第一次监听目标元素的时候
+- **目标**元素与设备视口或指定元素相交。在交叉观察器 API 中，指定元素被称为**根元素**或**根**。
+- 观察者（Observer）第一次监听观察目标元素。
```
Comment (jasonren0403) [inline files/zh-cn/web/api/intersection_observer_api/index.md:32]:
```suggestion
- 观察器（Observer）第一次监听观察目标元素。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 8]
```diff
@@ -49,36 +51,37 @@ let options = {
...
 ```
 
-阈值为 1.0 意味着目标元素完全出现在 root 选项指定的元素中可见时，回调函数将会被执行。
+阈值为 1.0 意味着目标元素完全出现在 `root` 选项指定的元素中 100% 可见时，回调函数将会被执行。
 
-#### Intersection observer options
+#### IntersectionObserver 选项
 
-传递到 {{domxref("IntersectionObserver.IntersectionObserver", "IntersectionObserver()")}} 构造函数的 `options` 对象，允许你控制观察者的回调函数的被调用时的环境。它有以下字段：
+传递到 {{domxref("IntersectionObserver.IntersectionObserver", "IntersectionObserver()")}} 构造函数的 `options` 对象，可以控制在什么情况下调用观察者的回调。它有以下字段：
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:58]:
```suggestion
传递到 {{domxref("IntersectionObserver.IntersectionObserver", "IntersectionObserver()")}} 构造函数的 `options` 对象，可以控制在什么情况下调用观察器的回调。它有以下字段：
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 9]
```diff
@@ -49,36 +51,37 @@ let options = {
...
 
 - `root`
-  - : 指定根 (**root**) 元素，用于检查目标的可见性。必须是目标元素的父级元素。如果未指定或者为`null`，则默认为浏览器视窗。
+  - : 用作视口的元素，用于检查目标的可见性。必须是目标的祖先。如果未指定或为 `null`，则默认为浏览器视口。
 - `rootMargin`
-  - : 根 (**root**) 元素的外边距。类似于 CSS 中的 {{cssxref("margin")}} 属性，比如 "`10px 20px 30px 40px"` (top、right、bottom、left)。如果有指定 root 参数，则 rootMargin 也可以使用百分比来取值。该属性值是用作 root 元素和 target 发生交集时候的计算交集的区域范围，使用该属性可以控制 root 元素每一边的收缩或者扩张。默认值为四个边距全是 0。
+  - : 根周围的边距。其值可以类似于 CSS {{cssxref("margin")}} 属性，例如 `"10px 20px 30px 40px"`（上、右、下、左）。这些值可以是百分比。在计算交叉点之前，这组值用于增大或缩小根元素边框的每一侧。默认值为全零。
 - `threshold`
-  - : 可以是单一的 number 也可以是 number 数组，target 元素和 root 元素相交程度达到该值的时候 IntersectionObserver 注册的回调函数将会被执行。如果你只是想要探测当 target 元素的在 root 元素中的可见性超过 50% 的时候，你可以指定该属性值为 0.5。如果你想要 target 元素在 root 元素的可见程度每多 25% 就执行一次回调，那么你可以指定一个数组 `[0, 0.25, 0.5, 0.75, 1]`。默认值是 0 (意味着只要有一个 target 像素出现在 root 元素中，回调函数将会被执行)。该值为 1.0 含义是当 target 完全出现在 root 元素中时候 回调才会被执行。
+  - : 一个数字或一个数字数组，表示目标可见度达到多少百分比时，观察者的回调就应该执行。如果只想在能见度超过 50% 时检测，可以使用 0.5 的值。如果希望每次能见度超过 25% 时都执行回调，则需要指定数组 \[0,0.25,0.5,0.75,1]。默认值为 0（这意味着只要有一个像素可见，回调就会运行）。值为 1.0 意味着在每个像素都可见之前，阈值不会被认为已通过。
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:65]:
```suggestion
  - : 一个数字或一个数字数组，表示目标可见度达到多少百分比时，观察器的回调就应该执行。如果只想在能见度超过 50% 时检测，可以使用 0.5 的值。如果希望每次能见度超过 25% 时都执行回调，则需要指定数组 \[0, 0.25, 0.5, 0.75, 1]。默认值为 0（这意味着只要有一个像素可见，回调就会运行）。值为 1.0 意味着在每个像素都可见之前，阈值不会被认为已通过。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 10]
```diff
@@ -90,40 +93,48 @@ let callback = (entries, observer) => {
...
+在创建 {{domxref("IntersectionObserver")}} 时，可以通过设置**根边距**（rootMargin）来进一步调整交叉点根矩形。`rootMargin` 中的值定义了添加到交叉点根边界框每一侧的偏移量，以创建最终的交叉点根边界（执行回调时将在 {{domxref("IntersectionObserverEntry.rootBounds")}} 中显示）。
 
-#### Thresholds
+#### 阈值
 
-IntersectionObserver API 并不会每次在元素的交集发生变化的时候都会执行回调。相反它使用了 **thresholds** 参数。当你创建一个 observer 的时候，你可以提供一个或者多个 number 类型的数值用来表示 target 元素在 root 元素的可见程序的百分比，然后，API 的回调函数只会在元素达到 **thresholds** 规定的阈值时才会执行。
+交叉观察器 API 使用**阈值**，而不是报告目标元素可见度的每一个微小变化。创建观察器时，可以提供一个或多个数值，代表目标元素可见度的百分比。然后，API 只报告超过这些阈值的可见性变化。
 
-例如，当你想要在 target 在 root 元素中中的可见性每超过 25% 或者减少 25% 的时候都通知一次。你可以在创建 observer 的时候指定 **thresholds** 属性值为 `[0, 0.25, 0.5, 0.75, 1]`，你可以通过检测在每次交集发生变化的时候的都会传递回调函数的参数 {{domxref("IntersectionObserverEntry.isIntersecting", "isIntersecting")}} 的属性值来判断 target 元素在 root 元素中的可见性是否发生变化。如果 `isIntersecting` 为真，target 元素的至少已经达到 **thresholds** 属性值当中规定的其中一个阈值，如果为假，则 target 元素不在给定的阈值范围内可见。
+例如，如果希望每次目标元素的可见度向后或向前越过每个 25% 的标记时都能得到通知，可以在创建观察者时指定数组 \[0,0.25,0.5,0.75,1] 作为阈值列表。
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:124]:
```suggestion
例如，如果希望每次目标元素的可见度向后或向前越过每个 25% 的标记时都能得到通知，可以在创建观察器时指定数组 \[0, 0.25, 0.5, 0.75, 1] 作为阈值列表。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 11]
```diff
@@ -320,37 +328,37 @@ intersectionCallback(entries) => {
...
-}
+};
 ```
 
-## Interfaces
+## 接口
 
 - {{domxref("IntersectionObserver")}}
-  - : The primary interface for the Intersection Observer API. Provides methods for creating and managing an observer which can watch any number of target elements for the same intersection configuration. Each observer can asynchronously observe changes in the intersection between one or more target elements and a shared ancestor element or with their top-level {{domxref("Document")}}'s {{Glossary('viewport')}}. The ancestor or viewport is referred to as the **root**.
+  - : 交叉观察器 API 的主要接口。它提供了创建和管理观察者的方法，观察者可以针对相同的交叉点配置观察任意数量的目标元素。每个观察者都可以异步观察一个或多个目标元素与共享的祖先元素或其顶层 {{domxref("Document")}} 的{{Glossary('viewport', '视口')}}之间的交集变化。祖先或视口被称为**根**。
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:337]:
```suggestion
  - : 交叉观察器 API 的主要接口。它提供了创建和管理观察器的方法，观察器可以针对相同的交叉点配置观察任意数量的目标元素。每个观察器都可以异步观察一个或多个目标元素与共享的祖先元素或其顶层 {{domxref("Document")}} 的{{Glossary('viewport', '视口')}}之间的交集变化。祖先或视口被称为**根**。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 12]
```diff
@@ -438,17 +446,17 @@ function createObserver() {
 }
 ```
 
-This begins by setting up an `options` object containing the settings for the observer. We want to watch for changes in visibility of the target element relative to the document's viewport, so `root` is `null`. We need no margin, so the margin offset, `rootMargin`, is specified as "0px". This causes the observer to watch for changes in the intersection between the target element's bounds and those of the viewport, without any added (or subtracted) space.
+首先要设置一个包含观察器设置的 `options` 对象。我们希望观察目标元素相对于文档视口的可见性变化，因此 `root` 为 `null`。我们不需要边距，因此边距偏移量 `rootMargin` 被指定为“0px”。这将导致观察者在不增加（或减少）空间的情况下观察目标元素边界与视口边界之间交点的变化。
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:449]:
```suggestion
首先要设置一个包含观察器设置的 `options` 对象。我们希望观察目标元素相对于文档视口的可见性变化，因此 `root` 为 `null`。我们不需要边距，因此边距偏移量 `rootMargin` 被指定为“0px”。这将导致观察器在不增加（或减少）空间的情况下观察目标元素边界与视口边界之间交点的变化。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 13]
```diff
@@ -49,36 +51,37 @@ let options = {
...
 observer.observe(target);
-```
 
-每当目标满足该 IntersectionObserver 指定的 threshold 值，回调被调用。
+// 我们为观察器设置的回调将在第一次执行，
+// 它将等待我们为观察器分配目标（即使目标当前不可见）
+```
 
-只要目标满足为 IntersectionObserver 指定的阈值，就会调用回调。回调接收 {{domxref("IntersectionObserverEntry")}} 对象和观察者的列表：
+每当目标满足该 `IntersectionObserver` 指定的 threshold 值，回调被调用。回调接收 {{domxref("IntersectionObserverEntry")}} 对象和观察器的列表：
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:79]:
```suggestion
每当目标满足该 `IntersectionObserver` 指定的阈值（threshold），回调被调用。回调接收 {{domxref("IntersectionObserverEntry")}} 对象和观察器的列表：
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 14]
```diff
@@ -49,36 +51,37 @@ let options = {
...
 
 ```js
 let callback = (entries, observer) => {
   entries.forEach((entry) => {
-    // Each entry describes an intersection change for one observed target element:
+    // 每个条目描述一个目标元素观测点的交叉变化：
     // entry.boundingClientRect
     // entry.intersectionRatio
     // entry.intersectionRect
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:87]: add spaces

image

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 15]
```diff
@@ -90,40 +93,48 @@ let callback = (entries, observer) => {
...
+例如，如果希望每次目标元素的可见度向后或向前越过每个 25% 的标记时都能得到通知，可以在创建观察者时指定数组 \[0,0.25,0.5,0.75,1] 作为阈值列表。
 
-Note that it's possible to have a non-zero intersection rectangle, which can happen if the intersection is exactly along the boundary between the two or the area of {{domxref("IntersectionObserverEntry.boundingClientRect", "boundingClientRect")}} is zero. This state of the target and root sharing a boundary line is not considered enough to be considered transitioning into an intersecting state.
+调用回调时，系统会接收一个 `IntersectionObserverEntry` 对象列表，每个观察到的目标都会有一个对象，这些目标与根相交的程度发生了变化，使得暴露量在任一方向上都超过了某个阈值。
 
-为了让我们感受下 thresholds 是如何工作的，尝试滚动以下的例子，每一个 colored box 的四个边角都会展示自身在 root 元素中的可见程度百分比，所以在你滚动 root 的时候你将会看到四个边角的数值一直在发生变化。每一个 box 都有不同的 thresholds：
+通过查看条目的 {{domxref("IntersectionObserverEntry.isIntersecting", "isIntersecting")}} 属性，可以了解目标当前是否与根相交；如果其值为 `true`，则表示目标至少与根元素或文档部分相交。这样就可以确定该条目是代表元素从相交到不再相交的过渡，还是代表从不相交到相交的过渡。
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:128]:
```suggestion
通过查看条目的 {{domxref("IntersectionObserverEntry.isIntersecting", "isIntersecting")}} 属性，可以了解目标*当前*是否与根相交；如果其值为 `true`，则表示目标至少与根元素或文档部分相交。这样就可以确定该条目是代表元素从相交到不再相交的过渡，还是代表从不相交到相交的过渡。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 16]
```diff
@@ -90,40 +93,48 @@ let callback = (entries, observer) => {
...
 
-- 第一个盒子的 thresholds 包含每个可视百分比，也就是说，{{domxref("IntersectionObserver.thresholds")}} 数组是 `[0.00, 0.01, 0.02, ..., 0.99, 1.00]`。
-- 第二个盒子只有唯一的值 `[0.5]`。
-- 第三个盒子的 thresholds 按 10% 从 0 递增 (0%, 10%, 20%, etc.)。
-- 最后一个盒子为 `[0, 0.25, 0.5, 0.75, 1.0]`。
+请注意，可能不存在相交矩形，如果交集正好沿着两者之间的边界，或者 {{domxref("IntersectionObserverEntry.boundingClientRect", "boundingClientRect")}} 的面积为零。这种目标和根共享边界线的状态不足以被视为过渡到相交状态。
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:130]:
```suggestion
请注意，可能存在零相交矩形，如果交集正好沿着两者之间的边界，或者 {{domxref("IntersectionObserverEntry.boundingClientRect", "boundingClientRect")}} 的面积为零。这种目标和根共享边界线的状态不足以被视为过渡到相交状态。
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 17]
```diff
@@ -90,40 +93,48 @@ let callback = (entries, observer) => {
...
+
+为了让我们感受下阈值是如何工作的，尝试滚动以下的盒子，每一个带颜色盒子的四个边角都会展示自身在根元素中的可见程度百分比，所以在你滚动根元素的时候你将会看到四个边角的数值一直在发生变化。每一个盒子都有不同的阈值：
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:132]:
```suggestion
要感受阈值是如何工作的，请尝试滚动下面的盒子，每一个带颜色盒子的四个边角都会展示自身在根元素中的可见程度百分比，所以在你滚动根元素的时候你将会看到四个边角的数值一直在发生变化。每一个盒子都有不同的阈值：
```

File: files/zh-cn/web/api/intersection_observer_api/index.md [hunk 18]
```diff
@@ -288,30 +296,30 @@ intersectionCallback = (entries) => {
 startup();
 ```
 
-{{EmbedLiveSample("Thresholds", 500, 500)}}
+{{EmbedLiveSample("阈值", 500, 500)}}
 
-#### Clipping and the intersection rectangle
+#### 剪切和相交矩形
```
Comment (yin1999) [inline files/zh-cn/web/api/intersection_observer_api/index.md:301]:
```suggestion
#### 裁剪和相交矩形
```

By the way, could you align the translation of "intersection rectangle"?

**Review by yin1999 — APPROVED** [review]
Thanks @jasonren0403

**Pull request merged**